### PR TITLE
Ptop fixes

### PR DESCRIPTION
**Pull request by snopoke**
@czue 

cc @NoahCarnahan 

:fishing_pole_and_fish:

### CODE REVIEW

File: corehq/pillows/user.py [hunk 1]
```diff
@@ -69,13 +69,14 @@ def get_unknown_users_pillow(pillow_id='unknown-users-pillow'):
...
+    change_feed = KafkaChangeFeed(topics=[FORM, FORM_SQL], group_id='unknown-users')
     return ConstructedPillow(
         name=pillow_id,
         checkpoint=checkpoint,
-        change_feed=KafkaChangeFeed(topics=[FORM, FORM_SQL], group_id='unknown-users'),
+        change_feed=change_feed,
         processor=processor,
-        change_processed_event_handler=PillowCheckpointEventHandler(
-            checkpoint=checkpoint, checkpoint_frequency=100,
+        change_processed_event_handler=MultiTopicCheckpointEventHandler(
```
Comment (czue) [inline corehq/pillows/user.py]: were these just broken somehow? i'm bummed that tests didn't catch it

File: corehq/apps/hqcase/management/commands/ptop_preindex.py [hunk 2]
```diff
@@ -33,7 +33,12 @@ def get_reindex_commands(alias_name):
         'hqgroups': [('ptop_reindexer_v2', {'index': 'group'})],
         'report_xforms': ['ptop_fast_reindex_reportxforms'],
         'report_cases': ['ptop_fast_reindex_reportcases'],
-        'case_search': [('ptop_reindexer_v2', {'index': 'case-search'})]
+        'case_search': [('ptop_reindexer_v2', {'index': 'case-search'})],
+        'ledgers': [
+            ('ptop_reindexer_v2', {'index': 'ledger-v1'}),
+            ('ptop_reindexer_v2', {'index': 'ledger-v2'}),
+        ],
```
Comment (czue) [inline corehq/apps/hqcase/management/commands/ptop_preindex.py]: will be interesting to see how these do. probably the biggest reindex to date on the new stuff

**Comment by czue** [comment]
:+1: on :white_check_mark: thanks for fixes

File: corehq/pillows/user.py (hunk 1 above)
Comment (snopoke) [inline corehq/pillows/user.py]: They work fine if they've already been converted to multicheckpoints but the checkpoint doesn't exist then it breaks.

Comment (snopoke) [inline corehq/pillows/user.py]: I'll take a look at the tests

Comment (czue) [inline corehq/pillows/user.py]: (no need to from my side if that's the failure mode - i was worried they were flat out broken)

**Comment by czue** [comment]
at least one of the test fails looks real:
```
======================================================================
FAIL: test_prod_config (testapps.test_elasticsearch.tests.test_prod_es_indices.ProdIndexManagementTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/testapps/test_elasticsearch/tests/test_prod_es_indices.py", line 27, in test_prod_config
    self.assertEqual(EXPECTED_PROD_INDICES, found_prod_indices)
AssertionError: Lists differ: [{'alias': 'case_search', 'typ... != [{'alias': u'case_search', 'ty...
First differing element 6:
{'alias': 'report_cases', 'type': 'report_case', 'index': 'test_report_cases_czei39du507m9mmpqk3y01x72a3ux4p0'}
{'alias': u'ledgers', 'type': u'ledger', 'index': u'test_ledgers_2016-03-15'}
Second list contains 1 additional elements.
First extra element 10:
{'alias': u'xforms', 'type': u'xform', 'index': u'test_xforms_2016-03-02'}
```

**Comment by czue** [comment]
this one does not look related, but also failed on https://github.com/dimagi/commcare-hq/pull/11646 so might be recently introduced:
```
======================================================================
FAIL: test_transfer_first_doc_order (corehq.apps.commtrack.tests.test_xml.CommTrackBalanceTransferTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/util/test_utils.py", line 230, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 203, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/corehq/util/test_utils.py", line 218, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 199, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/commtrack/tests/test_xml.py", line 398, in test_transfer_first_doc_order
    self.check_product_stock(self.sp, product, final, 0)
  File "/mnt/corehq/apps/commtrack/tests/test_xml.py", line 227, in check_product_stock
    self.assertEqual(expected_soh, latest_trans.stock_on_hand)
AssertionError: Decimal('50.0') != 150
```

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**